packages/expo-ui/ios/Modifiers/ViewModifierRegistry.swift

```diff
@@ -450,6 +450,58 @@ internal struct OnDisappearModifier: ViewModifier, Record {
   }
 }
 
+/**
+ * Manages refresh operation continuations for the refreshable modifier.
+ * This singleton stores continuations that are resumed when JavaScript signals completion.
+ */
+internal final class RefreshableManager {
+  static let shared = RefreshableManager()
+
+  private var continuations: [String: CheckedContinuation<Void, Never>] = [:]
+  private let queue = DispatchQueue(label: "expo.ui.refreshable")
+
+  private init() {}
+
+  func storeContinuation(_ continuation: CheckedContinuation<Void, Never>, for id: String) {
+    queue.async {
+      self.continuations[id] = continuation
+    }
+  }
+
+  func completeRefresh(id: String) {
+    queue.async {
+      if let continuation = self.continuations.removeValue(forKey: id) {
+        continuation.resume()
+      }
+    }
+  }
+}
+
+internal struct RefreshableModifier: ViewModifier, Record {
```

Comment:
can we move this (and RefreshableManager) to a separate file? we're starting to move new modifiers to a separate file as it is getting huge and also helps to prevent conflicts.